agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -0,0 +1,664 @@
+// Create is used to create all the fsxwindowsfileserver resources for a given task
+func (fv *FSxWindowsFileServerResource) Create() error {
+	var err error
+	var iamCredentials credentials.IAMRoleCredentials
+
+	executionCredentials, ok := fv.credentialsManager.GetTaskCredentials(fv.GetExecutionCredentialsID())
+	if !ok {
+		return errors.New("fsxwindowsfileserver resource: unable to find execution role credentials")
+	}
+	iamCredentials = executionCredentials.GetIAMRoleCredentials()
+
+	volumeConfig := fv.GetVolumeConfig()
+	err = fv.retrieveCredentials(volumeConfig.AuthConfig.CredentialsParameter, iamCredentials)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	err = fv.retrieveFileSystemDNSName(volumeConfig.FileSystemID, iamCredentials)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	fv.handleRootDirectory(volumeConfig.RootDirectory)
+
+	// formatting to keep powershell happy
+	remotePath := fmt.Sprintf(`\\%s\share`, fv.GetFileSystemDNSName())
+	if volumeConfig.RootDirectory != "" {
+		remotePath = fmt.Sprintf(`%s\%s`, remotePath, volumeConfig.RootDirectory)
+	}
+	localPath := volumeConfig.FSxWindowsFileServerHostPath
+
+	creds := fv.GetCredentials()
+	if creds.Username == "" || creds.Password == "" {
+		return errors.New("invalid credentials for mounting fsxwindowsfileserver volume on the container instance")
+	}
+
+	// formatting to keep powershell happy
+	username := volumeConfig.AuthConfig.Domain + `\` + creds.Username
+	password := creds.Password
+
+	err = fv.performHostMount(remotePath, localPath, username, password)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveCredentials(credentialsParameterARN string, iamCredentials credentials.IAMRoleCredentials) error {
+	parsedARN, err := arn.Parse(credentialsParameterARN)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	parsedARNService := parsedARN.Service
+	if parsedARNService == "ssm" {
+		err = fv.retrieveSSMCredentials(credentialsParameterARN, iamCredentials)
+		if err != nil {
+			seelog.Errorf("Failed to retrieve credentials from ssm: %v", err)
+			fv.setTerminalReason(err.Error())
+			return err
+		}
+	} else if parsedARNService == "secretsmanager" {
+		err = fv.retrieveASMCredentials(credentialsParameterARN, iamCredentials)
+		if err != nil {
+			seelog.Errorf("Failed to retrieve credentials from asm: %v", err)
+			fv.setTerminalReason(err.Error())
+			return err
+		}
+	} else {
+		err = errors.New("unsupported credentialsParameter ARN, only ssm/secretsmanager ARNs are valid")
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveSSMCredentials(credentialsParameterARN string, iamCredentials credentials.IAMRoleCredentials) error {
+	parsedARN, err := arn.Parse(credentialsParameterARN)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
```

Comment:
this is being set again in the parent method, this is not needed(there are many other instances of this)

Reply:
will remove